saw-core/src/Verifier/SAW/SharedTerm.hs

```diff
@@ -1145,23 +1178,31 @@ scTupleSelector sc t i n
                    scTupleSelector sc t' (i - 1) (n - 1)
   | otherwise = fail "scTupleSelector: non-positive index"
 
+-- | Create a term representing the type of a function, given a parameter and
+-- result type (as terms).
 scFun :: SharedContext -> Term -> Term -> IO Term
 scFun sc a b = do b' <- incVars sc 0 1 b
                   scTermF sc (Pi "_" a b')
 
+-- | Create a term representing the type of a function, given a list of
+-- parameter types and a result type (as terms).
 scFunAll :: SharedContext
          -> [Term]
          -> Term
          -> IO Term
 scFunAll sc argTypes resultType = foldrM (scFun sc) resultType argTypes
 
+-- | Create a lambda term from a parameter name, parameter type (as a term),
+-- and a body.
 scLambda :: SharedContext
          -> String
          -> Term
          -> Term
          -> IO Term
 scLambda sc varname ty body = scTermF sc (Lambda varname ty body)
 
+-- | Create a lambda term of multiple arguments (curried) from a list
+-- associating parameter names to types (as terms) and a body.
```

Comment:
As above, we should mention conventions for de Bruijn indices. Index 0 in the body refers to the last argument in the list, and index `n-1` refers to the first.